### ISSUE TITLE
Wildcard resource pattern doesn't work with Gradle plugin

### ISSUE THREAD

**Issue by Skater901** [graalvm/native-build-tools#639]
**Before reporting**

- This repository should be used to report **issues on the Maven or Gradle plugins for GraalVM**.
- Please report issues which are specific to [the Spring Framework](https://spring.io/) or [the Micronaut framework](https://micronaut.io/) to their specific repositories.
- Do not report issues with building your specific application, e.g errors which happen at image build time like classes initialized at build time, or missing classes as run time: those are not related to the plugins but problems with configuration. You can refer to the [GraalVM native image documentation](https://www.graalvm.org/latest/reference-manual/native-image/) for available options and the [plugins documentation](https://graalvm.github.io/native-build-tools) for how to use them with the plugin.

**Describe the bug**
When trying to include resources from a Jar in the project dependencies, using a wild card pattern doesn't work to include the files. Only specifying the files explicitly by name ensures they are included in the native image.
**Make sure that you have read [the documentation](https://graalvm.github.io/native-build-tools) and that you are using the latest plugin version.**

**To Reproduce**

A reproducible example can be found [here](https://github.com/Skater901/native-resources-test)

```kotlin
dependencies {
    implementation("org.webjars:swagger-ui:5.17.11")
}

graalvmNative {
    binaries.named("main") {
        resources.includedPatterns.add("META-INF/resources/webjars/swagger-ui/5.17.11/*")
    }
}
```
```bash
$ ./gradlew nativeCompile
$ ./build/native/nativeImage/{binary-name}
```
Please use backticks to [properly format code](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting).
If possible please attach a complete reproducer here (either as [a zip file](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) or as a link to public repository/branch).

**Expected behavior**
All files in the META-INF/resources/webjars/swagger-ui/5.17.11 folder from the `org.webjars:swagger-ui:5.17.11` library should be included in the produce native image.

**Logs**
Add logs to help explain your problem. 
Please use backticks to [properly format big logs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks). Example:
````
```
[thread:1] scope: main
  [thread:1] scope: main.registerResource
  ResourcesFeature: registerResource: java.base:java/lang/uniName.dat
  [thread:1] scope: main.registerResource
  ResourcesFeature: registerResource: META-INF/resources/webjars/swagger-ui/5.17.11
```
````

**System Info (please complete the following information):**
 - OS: Windows 10
 - GraalVM Version `Oracle GraalVM 21.0.5+9.1`
 - Java Version `21.0.5+9-LTS`
 - Plugin version `id("org.graalvm.buildtools.native") version "0.10.3"`

**Additional context**

**Comment by dnestoro** [comment]
@melix are you familiar with this way of registering resources:
```
graalvmNative {
    binaries.named("main") {
        resources.includedPatterns.add("META-INF/resources/webjars/swagger-ui/5.17.11/*")
    }
}
```

**Comment by Skater901** [comment]
I forgot to include the documentation I was looking at: https://www.graalvm.org/latest/reference-manual/native-image/metadata/#resource-metadata-in-json

For what it's worth, the docs for GraalVM for JDK 21 (which is what I'm using) don't talk about using wildcards, but they do include examples of patterns with wildcards: https://www.graalvm.org/jdk21/reference-manual/native-image/metadata/#resource-metadata-in-json

I'm assuming these docs are applicable to the Gradle plugin, but if they're not, then this could simply be a case of insufficient documentation for the Gradle plugin, and I haven't correctly configured the resource inclusion pattern for the Gradle plugin.

**Comment by melix** [comment]
So when you add the includes/excludes in the DSL block like this:

```
graalvmNative {
    binaries.named("main") {
        resources.includedPatterns.add("META-INF/resources/webjars/swagger-ui/5.17.11/*")
    }
}
```

what will happen is that the generated resources configuration file will simply have an additional include pattern:

```
{
  "resources": {
    "includes": [
      {
        "pattern": "META-INF/resources/webjars/swagger-ui/5.17.11/*"
      }
    ],
    "excludes": []
  },
  "bundles": []
}
```

This file is used during compilation, so if the resource is missing, I would suspect a different problem. There's nothing specific to Gradle here, it's simply copying the pattern to the configuration file.

**Comment by Skater901** [comment]
Sure, it's entirely possible that this is a bug with the native image builder itself, and is unrelated to the Gradle plugin, but I'm using the Gradle plugin so that's what I reported it as. 😛 Since I didn't know if I could reproduce it without the plugin.

Your theory sounds correct that it's probably not the plugin, but do you know what the issue is, then? Is it a problem with the actual native image builder? Does GraalVM 21 not support wild cards for resource patterns, or something?

**Comment by dnestoro** [comment]
**TLDR;**

Generated file corresponds to [this json schema](https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/assets/resource-config-schema-v1.0.0.json) which uses `regex` and not `glob`. Globs have been introduced later (as of GraalVM for JDK22 I think), and the file format which supports globs corresponds to [this json schema](https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/assets/resource-config-schema-v1.1.0.json).
Therefore if this DSL block just generates config file in the old format, you can't use globs there.

**Detailed explanation**

> Your theory sounds correct that it's probably not the plugin, but do you know what the issue is, then?

The problem here is the following:

- if this block (as @melix said):
```
graalvmNative {
    binaries.named("main") {
        resources.includedPatterns.add("META-INF/resources/webjars/swagger-ui/5.17.11/*")
    }
}
```

- generates this block:
```
{
  "resources": {
    "includes": [
      {
        "pattern": "META-INF/resources/webjars/swagger-ui/5.17.11/*"
      }
    ],
    "excludes": []
  },
  "bundles": []
}
```
then you can't use glob as a pattern. In this format of `resource-config.json` you must use `regex` as a pattern and not `glob` (see [this](https://www.graalvm.org/jdk21/reference-manual/native-image/dynamic-features/Resources/))

Globs have been introduced later, and the file format that accepts them is described [here](https://www.graalvm.org/latest/reference-manual/native-image/metadata/#resource-metadata-in-json).

**Comment by Skater901** [comment]
Yeah cool, I get it. Changing the pattern to `"^META-INF/resources/webjars/swagger-ui/5.17.11/.*"` makes it work. Not a bug, just reading the docs for the wrong version. :) Thank you very much for taking the time to figure this one out for me!
